test/logflare_web/controllers/endpoints_controller_test.exs

```diff
@@ -109,6 +109,52 @@ defmodule LogflareWeb.EndpointsControllerTest do
     end
   end
 
+  describe "bigquery with labels" do
+    setup do
+      _plan = insert(:plan, name: "Free")
+      user = insert(:user)
+      source = build(:source, user: user)
+      {:ok, user: user, source: source}
+    end
+
+    test "reference params in label, my_label=@my_param", %{
+      conn: conn,
+      user: user,
+      source: source
+    } do
+      pid = self()
+
+      GoogleApi.BigQuery.V2.Api.Jobs
+      |> stub(:bigquery_jobs_query, fn _conn, _proj_id, opts ->
+        send(pid, opts[:body].labels)
+        {:ok, TestUtils.gen_bq_response()}
+      end)
+
+      endpoint =
+        insert(:endpoint,
+          user: user,
+          enable_auth: true,
+          query: "with a as (select 1 as b) select b from a",
+          labels: "my_label=@my_param,other_value"
+        )
+
+      conn =
+        conn
+        |> put_req_header("x-api-key", user.api_key)
+        |> put_req_header("content-type", "application/json")
+        |> put_req_header("lf-endpoint-labels", "other_value=1234,omit=334")
+        |> get(
+          ~p"/api/endpoints/query/#{endpoint.name}?#{%{"other_value" => 1234, "omit" => 333, "my_param" => "my_value", "sql" => "select 2"}}"
+        )
+
+      assert [_] = json_response(conn, 200)["result"]
+      assert conn.halted == false
+      assert_received labels = %{"my_label" => "my_value", "other_value" => "1234"}
+      refute labels["omit"]
+      refute labels["sql"]
```

Comment:
nitpick: maybe worth asserting on all the possible label keys, then we don't need to do refutations:

```suggestion
      assert_received labels
      assert labels == %{"my_label" => "my_value", "other_value" => "1234"}
```

but yeah, totally fine is!